client/src/main/scala/drt/client/services/SPACircuit.scala

```diff
@@ -330,40 +330,57 @@ object HandyStuff {
   extends LoggingActionHandler(modelRW) {
   protected def handle = {
     case RunAllSimulations() =>
-      log.info(s"run simulation for ${staffDeployments.value}")
+      log.info(s"RunAllSimulations")
       val actions = staffDeployments.value.flatMap {
         case (terminalName, queueMapPot) => {
           queueMapPot.map {
             case (queueName, deployedDesks) =>
-              //              val queueWorkload = getTerminalQueueWorkload(terminalName, queueName)
               val desks = deployedDesks.get.items.map(_.deskRec)
               val desksList = desks.toList
               Effect(Future(RunSimulation(terminalName, queueName, desksList)))
           }
         }
       }.toList
-      log.info(s"runAllSimulations effects ${actions}")
+      effectOnly(seqOfEffectsToEffectSeq(actions))
+    case RunTerminalSimulations(terminalName) =>
+      log.info(s"RunTerminalSimulation for $terminalName")
+      val actions = staffDeployments.value.getOrElse(terminalName, Map()).map {
+        case (queueName, deployedDesks) =>
+          val desks = deployedDesks.get.items.map(_.deskRec)
+          val desksList = desks.toList
+          Effect(Future(RunSimulation(terminalName, queueName, desksList)))
+      }.toList
+      log.debug(s"RunTerminalSimulations effects ${actions}")
       effectOnly(seqOfEffectsToEffectSeq(actions))
     case rs@RunSimulation(terminalName, queueName, desks) =>
-      log.info(s"Requesting simulation for $terminalName, $queueName")
-      val queueWorkload = getTerminalQueueWorkload(terminalName, queueName)
-      val firstNonZeroIndex = queueWorkload.indexWhere(_ != 0)
-      log.info(s"run sim first != 0 workload $firstNonZeroIndex")
-      log.info(s"run sum $rs workload $queueWorkload")
-      val simulationResult: Future[SimulationResult] = AjaxClient[Api].processWork(terminalName, queueName, queueWorkload, desks).call()
-      effectOnly(
-        Effect(simulationResult.map(resp => UpdateSimulationResult(terminalName, queueName, resp)))
-      )
+      if (rawShifts.value.isReady && rawFixedPoints.value.isReady && movements.value.isReady && allQueueCrunchesReceived()) {
+        log.info(s"Requesting simulation for $terminalName, $queueName")
+        val queueWorkload = getTerminalQueueWorkload(terminalName, queueName)
+        val firstNonZeroIndex = queueWorkload.indexWhere(_ != 0)
+        log.debug(s"run sim first != 0 workload $firstNonZeroIndex")
+        log.debug(s"run sum $rs workload $queueWorkload")
+        val simulationResult: Future[SimulationResult] = AjaxClient[Api].processWork(terminalName, queueName, queueWorkload, desks).call()
+        effectOnly(
+          Effect(simulationResult.map(resp => UpdateSimulationResult(terminalName, queueName, resp)))
+        )
+      } else {
+        log.info(s"Not running simulation for $terminalName/$queueName. Shifts, Fixed points & movements not all ready yet")
+        noChange
```

Comment:
Shouldn't be required as each of the bits of data that are needed for a simulation now triggers one when they come through